docs/source/models.rst

```diff
@@ -139,48 +139,134 @@ For more information, see the :mod:`mlflow.pyfunc` documentation.
+also contain the ``python_function`` flavor, allowing them to be interpreted as generic
+Python functions for inference via :py:func:`mlflow.pyfunc.load_pyfunc()`. Finally, the
+:py:func:`mlflow.keras.load_model` function can be used to load MLflow models with the
+``keras`` flavors as `Keras Model objects <https://keras.io/models/about-keras-models/>`_.
+
+For more information, see :py:mod:`mlflow.keras`.
 
 MLeap (``mleap``)
 ^^^^^^^^^^^^^^^^^
 
-The ``mleap`` model flavor supports saving models using the MLeap persistence mechanism. A companion module for loading MLflow models with the MLeap flavor format is available in the ``mlflow/java`` package. For more information, see :py:mod:`mlflow.mleap`.
+The ``mleap`` model flavor supports saving Spark models in MLflow format using the 
+`MLeap <http://mleap-docs.combust.ml/>`_ persistence mechanism. MLeap is an inference-optimized 
+format and execution engine for Spark models that does not depend on 
+`SparkContext <https://spark.apache.org/docs/latest/api/python/pyspark.html#pyspark.SparkContext>`_
+to evaluate inputs.
+
+Spark models can be saved with in MLflow format with MLeap by specifying the ``sample_input`` 
+argument of the :py:func:`mlflow.spark.save_model()` or :py:func:`mlflow.spark.log_model()` method 
+(recommended). The :py:mod:`mlflow.mleap` module also defines 
+:py:func:`save_model() <mlflow.mleap.save_model>` and 
+:py:func:`log_model() <mlflow.mleap.log_model>` methods for saving MLeap models in MLflow format, 
+but these methods do not include the ``python_function`` flavor in the models they produce.
+
+A companion module for loading MLflow models with the MLeap flavor format is available 
+in the ``mlflow/java`` package. 
+
+For more information, see :py:mod:`mlflow.spark`, :py:mod:`mlflow.mleap`, and the
+`MLeap documentation <http://mleap-docs.combust.ml/>`_.
 
 PyTorch (``pytorch``)
 ^^^^^^^^^^^^^^^^^^^^^
 
-The ``pytorch`` model flavor enables logging and loading PyTorch models. Model is completely stored in `.pth` format using `torch.save(model)` method. Given a directory containing a saved model, you can log the model to MLflow via ``log_saved_model``. The saved model can then be loaded for inference via ``mlflow.pyfunc.load_pyfunc()``. For more information, see :py:mod:`mlflow.pytorch`.
+The ``pytorch`` model flavor enables logging and loading PyTorch models.
+
+The :py:mod:`mlflow.pytorch` module defines utilities for saving and loading MLflow models with the
+``pytorch`` flavor. The :py:func:`mlflow.pytorch.save_model()` and
+:py:func:`mlflow.pytorch.log_model()` methods can be used to save PyTorch models in MLflow format;
+both of these functions use the
+`torch.save() <https://pytorch.org/docs/stable/torch.html#torch.save>`_ method to serialize PyTorch
+models. Additionally, the :py:func:`mlflow.pytorch.load_model()` method can be used to load MLflow
+models with the ``pytorch`` flavor as PyTorch model objects. Finally, models produced by
```

Comment:
the :py:func:`mlflow.pytorch.load_model()` method can be used to load MLflow
models with the ``pytorch`` flavor -> You can use the :py:func:`mlflow.pytorch.load_model()` method to load MLflow models with the ``pytorch`` flavor

Reply:
Fixed!